### DISCUSSION TITLE
Ideas & Suggestions

### DISCUSSION THREAD

**Discussion by rednblkx** [rednblkx/HomeKey-ESP32#16] [Ideas]
You can share your ideas and suggestions here for the project here.

**Comment by jk2g** [comment]
Bom primeiramente parabens pelo projeto!!!
gostaria que se possivel add possibilidade de adicionar um sensor para status de aberto e fechado . um sensor magnetico add a uma gpio.obrigado desde ja .

**Reply by jk2g** [reply]
digo sobre o sensor pois para o fechamento automatico somente criando automaçôes .com a possibilidade de adicionar um sensor o status mudaria automatico quando sensor ABERTO/FECHADO.sem uso de automaçâo.

**Comment by rhamblen** [comment]
OK so I've got it all working.
Im using not red to unlock the door.  I was using a web ui for the lock but it was slow, so I've switched to MQTT commands and it is much quick, ie instantaneous!

Suggestion...
in the interface we can already send MQTT commands, but could we also set a custom topic and payload which is sent in addition.  Something which is sent when the door unlock is issued.   
This would be awesome as we then dont need a MIM line nodered to unlock the door.

**Reply by rednblkx** [reply]
There already is [MQTT Custom States](https://github.com/rednblkx/HomeKey-ESP32/wiki/MQTT#111-mqtt-custom-states)
They are tied to the internal lock states and only limited to numbers so it’s not complete custom payloads.
It was originally implemented in order to publish data directly to a Lock that has an MQTT API like Nuki’s.
The wiki needs a little work, let me know if you have any questions.

**Comment by konradhuebner** [comment]
@rednblkx there is another really nice project: https://github.com/esprfid/esp-rfid and I was wondering if a best-of-both would make sense:
- add user management for standard RFID to this project so that not only HomeKey works, but also the cheap tags. My kids don't have iPhones yet ;)
- maybe make use of some of their web-ui and infrastructure for this
- fully emulate a lock, and not only the reading. I want to integrate this into my garage door, and just need a 1s activation of the relais, and then I would be good again.

Ultimately, one could even add a fingerprint reader, for that there is also a project: https://github.com/frickelzeugs/FingerprintDoorbell

Would be happy to hear your thoughts :)

**Reply by electric-m** [reply]
I appreciate all of you ideas, but that seems like a huge change to the general project.

I do User management and the opening closing of the lock in nodered.

That way, so have Homekeys, Traditional Tags via NodeRED, and the correct animation and status of the locks also via NodeRED.

https://github.com/user-attachments/assets/cab8b298-963d-4aeb-8297-1df1bbf47c82

**Reply by rednblkx** [reply]
Hi @konradhuebner 
Appreciate your suggestions but I have to say that some of them go beyond the scope of this project unfortunately. 
The main focus for this project is and will be homekey, the basic scanning of tags was included because it felt silly not to, since it was already halfway there. Also, the addition of all this extra rfid features would make the firmware size go overboard and would prompt people to search for a board with 8MB or even 16MB. I realise i should’ve been more clear in the description that the purpose of this project is to provide homekey functionality, anything extra are only QoL additions.

However, not all hope is lost, i actually developed a HomeKit component for ESPHome and a HomeKey integration for it that can be found [here](https://github.com/rednblkx/HAP-ESPHome), so people can have this choice of customisation and expansion.

Additionally, I am actually planning on adding some simple uid tag whitelisting so you don’t have to rely on external validation.
Regarding your relay suggestion, you can use the success event, timeout can’t be changed as there is a bug but the default is 1s, however this won’t sync with the state in HomeKit for that I added GPIO actions but misses momentary state which will be added in the next version(present right now in the dev branch).

Again, thank you for the suggestions, let me know if you have any questions and you can also join the discord server for a chat if you want.

**Comment by KristianMuc** [comment]
Hi
First of all great project and thanks al lot for all the work.
One idea or question.
When you have added the digital key to the apple wallet it is actually not possible to share the key with someone else. This is for example possible with event tickets or also for my digital key from my car.
So far i understood it is only possible to use iphones of different useres as a key when you have a homekit zentrale and you add all the useres to your home. 
It could make sense to share the wallet key unlimeted or for some time with other persons you wont add to your complete home in apple home.
Is it possible to add this share function in the (wallet) key like for event tickets?

**Reply by rednblkx** [reply]
Hi there!

Yeah, sharing functionality would be nice but Apple hasn't implemented anything in this regard for Homekey which means there isn't really anything I can do from my side. 

However, they did announce Guest Access at WWDC24 that allows to share a key with someone not in your home, which is yet to arrive as it is packaged along with the new standard Aliro which hopefully will release with iOS 18.5

**Comment by thomasdittrich** [comment]
Hi guys, I think it would be good if you could use the ESP32 as a BLE proxy.
